pkg/reconciler/instances/istio/README.md

```diff
@@ -4,6 +4,33 @@
 
 Istio Reconciler manages the [Istio](https://github.com/kyma-project/kyma/tree/main/resources/istio-configuration) Kyma component. We support two latest minor Kyma 2.x releases and the `main` Kyma version.
 
+## Architecture
+
+Reconciliation in Kyma is handled by Reconciler. The Mothership Reconciler knows the reconciliation status of every managed Kyma cluster and initiates reconciliation of all Kyma components.
+
+The reconciliation of Istio is split into two different Kyma components: Istio and Istio Resources. The separation enables having the pure installation process of Istio under control via Istio reconciler implementation and using the Reconciler framework functionality of simply applying charts to the cluster for Istio Resources.
+
+The Istio component installs Istio on a Kyma cluster. For installation purposes, it requires Reconciler that uses `istioctl` and a rendered `istio-operator.yaml` file with Kyma-specific configurations. After proper installation/upgrade of the Istio service mesh within the cluster, the Istio proxies are restarted if needed. Within the whole Kyma reconciliation process, the installation of Istio is a prerequisite as other components rely on proper service mesh setup and on the installed CRDs.
+
+The Istio Resources component provides charts for additional resources that are related to Istio itself but are not related to the installation process. The resources:
+
+- kyma-gateway (istio-ingressgateway)
+- monitoring-related Grafana dashboards to monitor the Istio service mesh components
+
+See the diagram for details:
+
+![Istio Reconciler](./assets/istio-reconciler.svg)
+
+The following activity diagram presents detailed steps of the Istio installation/update process.
+
+![Istio Reconciliation](./assets/istio-reconciliation-action.svg)
+
+The reconciliation is executed by the Istio Reconciler. First, it verifies, using the rules explained in the diagram, if the Istio version found on the cluster and the Client version (istioctl) match. If the versions are compatible, either an installation or update process is triggered. Before the update, the version from the Istio [`values.yaml`](https://github.com/kyma-project/kyma/blob/main/resources/istio-configuration/values.yaml) is compared with the cluster version.
+
+If a customer makes changes in the Istio configuration that are not compatible with the Kyma setup configured within `istio-operator.yaml`, the Istio Reconciler automatically overwrites them with the default values.
+
+After choosing the proper Istio version for installation and applying back the default values, the Istio Reconciler patches the Istio Webhook to base on Kyma and Gardener assumptions.
+
```

Comment:
I'm wondering if we should include the Architecture section at the very end of the document and treat it as Details? 🤔